designs/extensible-cli-and-scaffolding-plugins-phase-1.md

```diff
@@ -135,6 +113,36 @@ type Deprecated interface {
 }
 ```
 
+## Configuration
+
+### Config version `3-alpha`
+
+Any changes that break `PROJECT` file backwards-compatibility require a version
+bump. This new version will be `3-alpha`, which will eventually be bumped to
+`3` once the below config changes have stabilized.
+
+### Project file plugin `layout`
+
+The `PROJECT` file will specify what base plugin generated the project under
+a `layout` key. `layout` will have the format: `Plugin.Name() + "/" + Plugin.Version()`.
+`version` and `layout` have versions with different meanings: `version` is the
+project config version, while `layout`'s version is the plugin semantic version.
+The value in `version` will determine that in `layout` by a plugin's supported
+project versions (via `SupportedProjectVersions()`).
+
+Example `PROJECT` file:
+
+```yaml
+version: "3-alpha"
```

Comment:
Hi @estroz,

The kubebuilder has the release tags and version projects. 
I understand that the plugin implementation will NOT break the layout of the project built with and the tag 4.0.0 with the plugins will still be working on with the built projects. So, IHMO the project version should still 2, otherwise, we need to create the V3 structure of files, and testdata mock data, book-v3 and etc ... 

@mengqiy @DirectXMan12 wdyt?